### PR TITLE
test: create helper silent auth token extractor

### PR DESCRIPTION
**Pull request by danielgent**
On the one hand _no logic in your tests_

On the other hand, is this logic? :smile:   I don't think it is. It's parsing... :thinking: 

I figure to catch all our errors given the complexity of this project...

We should test silent auth *Everywhere*

There's always the potential to get this wrong, and silent auth has been a contintual source of bugs for us :zipper_mouth_face:  (Micke seemed satisfied with auth today so the less the said the better)

### CODE REVIEW

File: integration-test/helpers/silent-auth.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  const silentAuthIdTokenPayload = parseJwt(silentAuthIdToken);
+
+  const silentAuthToken = iframeResponseJSON.access_token;
+
+  const silentAuthTokenPayload = parseJwt(silentAuthToken);
+
+  // these aren't typed! coming out as anys...
+  return {
+    accessToken: silentAuthTokenPayload,
+    idToken: silentAuthIdTokenPayload,
```
Comment (danielgent) [inline integration-test/helpers/silent-auth.ts:62]: maybe better if at least come out as `{ [key: string]: number };`

File: integration-test/flows/code-flow.spec.ts [hunk 2]
```diff
@@ -123,56 +124,16 @@ describe("code-flow", () => {
...
-    const iframeResponseJSON = JSON.parse(
-      responseBody.replace("response: ", ""),
+    const {
+      accessToken: silentAuthAccessTokenPayload,
+      idToken: silentAuthIdTokenPayload,
+    } = await doSilentAuthRequestAndReturnTokens(
+      setCookiesHeader,
+      worker,
+      nonce,
+      "clientId",
```
Comment (danielgent) [inline integration-test/flows/code-flow.spec.ts:134]: I think you'll agree this makes the tests a lot cleaner :sunglasses:

File: integration-test/flows/silent-auth.spec.ts [hunk 3]
```diff
@@ -78,54 +80,34 @@ describe("silent-auth", () => {
     expect(tokenResponse.status).toBe(302);
     expect(await tokenResponse.text()).toBe("Redirecting");
 
-    const authCookieHeader = tokenResponse.headers.get("set-cookie")!;
+    const setCookieHeader = tokenResponse.headers.get("set-cookie")!;
```
Comment (danielgent) [inline integration-test/flows/silent-auth.spec.ts:83]: arguably it wasn't worth using the util on this test suite as we're not using the parsed tokens but :shrug:

**Review by markusahlstrand — APPROVED** [review]
LGTM!

**Pull request merged**